### PR TITLE
Closing order fix and new pending orders list

### LINKED ISSUE #11: createOrder doesn't push order to this.orders until it is executed
I had this trouble with onTick, i has a condition that refers to this.orders.length

`if (tick.h > this.maShort1 && this.prevTick.h < this.maShort1 && this.orders.length < 1) {
    await this.createOrder(OrderType.SELL);
}`

it can create order on every tick, until the first order is executed, personally i had 7 orders instead of 1

### LINKED ISSUE #12: Infinite loop on await this.closeAll
i have this condition called on every tick, it crashes with heap out of memory, it might be caused by this issue  #https://github.com/debut-js/Core/issues/11
`if (
            (tick.c > this.maValue && this.prevTick.c < this.maValue) ||
            (tick.c < this.maValue && this.prevTick.c >  this.maValue)
        ) {
            console.log('Closing all orders');
            await this.closeAll();
        }
)`

### PR DESCRIPTION
**Pull request by BusinessDuck**
- fix #11 
- fix #12

### CODE REVIEW

File: src/modules/debut.ts [hunk 1]
```diff
@@ -309,6 +315,31 @@ export abstract class Debut implements DebutCore {
...
+     * Replace pending order to executed by cid
+     */
+    private replacePendingOrder(order: ExecutedOrder) {
+        const idx = this.orders.findIndex((item) => item.cid === order.cid);
+
+        if (idx !== -1) {
+            this.orders[idx] = order;
+        } else {
+            // TODO: Remove when fine
+            console.warn('Unkndown order for replace', this.orders, order);
```
Comment (StaRenn) [inline src/modules/debut.ts:328]: Очепятка

Comment (BusinessDuck) [reply src/modules/debut.ts]: fixed

**Comment by BusinessDuck** [comment]
I make a little test with strategies between master and this branch and result looks identical. good news

**Comment by BusinessDuck** [comment]
@StaRenn waiting for you feedback bout issues and its can be merged. 
2.4.3-beta.1 - you issues fix
2.4.3-beta.2 - all branch updates

**Pull request merged**